src/HotChocolate/Core/src/Types.CursorPagination/Connection.cs

```diff
@@ -54,7 +54,7 @@ public class Connection : IPage
         ConnectionPageInfo info,
         int totalCount = 0)
     {
-        _getTotalCount = _ => new(totalCount);
+        _getTotalCount = _ => new(info.TotalCount ?? totalCount);
```

Comment:
We actually never used total count before. I know use either total count of the connection page info or the one provided directly through the connection.